app.py

```diff
@@ -1,26 +1,403 @@
+    """
+    Only allow given plots
+    """
+    plots = ["bar", "line", "pie"]
+    try:
+        val = str(val).lower()
+    except ValueError:
+        raise argparse.ArgumentTypeError(f"{val} could not be parsed to a string")
+
+    if val not in plots:
+        raise argparse.ArgumentTypeError(f"{val} is not a valid plot type. Possible values: {', '.join(plots)}")
+    return val
+
+
+def _restricted_hashtags(val: str):
+    """
+    Only allow hashtags that follow our standards.
+    Typically we remove #-symbols even if a user added them.
+    Used for hashtags arg
+    """
+    try:
+        val = str(val).lower()
+    except ValueError:
+        raise argparse.ArgumentTypeError(f"{val} could not be parsed to a string")
+
+    val = re.sub(_REGEX_CHAR_MATCHER_HASHTAGS, "", val)
+    return val
+
+
+def _restricted_search_hashtags(val: str):
+    """
+    Only allow hashtags that follow our standards.
+    Typically we add a # even if user forgot them.
+
+    Important to use .lower to match program output.
+
+
+    Used for search_hashtags arg
+    """
+    try:
+        val = str(val).lower()
+    except ValueError:
+        raise argparse.ArgumentTypeError(f"{val} could not be parsed to a string")
+
+    if not val.startswith('#'):
+        return '#' + val
+    return val
+
+
+def _restricted_search_mentions(val: str):
+    """
+    Only allow mentions that follow our standards.
+    Typically we add @-symbol even if an user forgot them.
+
+    important to keep structure to match program output.
+
+    Used for search_mentions arg
+    """
+    try:
+        val = str(val)
+    except ValueError:
+        raise argparse.ArgumentTypeError(f"{val} could not be parsed to a string")
+
+    if not val.startswith('@'):
+        return '@' + val
+    return val
+
+
+#########CUSTOM TYPES#########
+
+#########HELPER METHODS#########
+
+class CustomFormatter(argparse.ArgumentDefaultsHelpFormatter, argparse.RawTextHelpFormatter):
+    """
+    Allows us to use two formatters for argparse instead of only the default 1 (hackish)
+    https://stackoverflow.com/a/18462760
+    """
+
+    def _split_lines(self, text, width):
+        # https://stackoverflow.com/a/29498894
+        # modify super to add newline between arguments
+        lines = super()._split_lines(text, width)
+        if text.endswith('\n'):  # custom newline
+            lines += ['']
+        else:
+            lines += ['']  # also add newline between different arguments
+        return lines
+
+
+def _default_dates():
+    """
+    Generates today and five days from now for use with default values of the date-argument
+    """
+    today = datetime.now().date()
+    five_days_from_now = today + timedelta(days=5)
+    # create readable format, as should be input
+    # return [today.strftime('%Y-%m-%d'), five_days_from_now.strftime('%Y-%m-%d')]
+    return [today, five_days_from_now]
+
+
+def _filter_search_values(key: str, values: list, collection: list):
```

Comment:
Originally wanted to use get_by_key_value but python Filter does not allow handling many values:  
https://stackoverflow.com/a/4778869